src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -442,4 +498,95 @@ private Model GetVocabularyDictionary()
             }
         }
     }
+
+    public sealed class WordEmbeddingsEstimator : TrivialEstimator<WordEmbeddingsTransform>
+    {
+        public WordEmbeddingsEstimator(IHostEnvironment env, string inputColumn, string outputColumn,
+           WordEmbeddingsTransform.PretrainedModelKind? modelKind = WordEmbeddingsTransform.PretrainedModelKind.Sswe)
+            : this(env, new WordEmbeddingsTransform(env, inputColumn, outputColumn, modelKind))
+        {
+        }
+
+        public WordEmbeddingsEstimator(IHostEnvironment env,
+            WordEmbeddingsTransform.PretrainedModelKind? modelKind = WordEmbeddingsTransform.PretrainedModelKind.Sswe,
+            string customLookupTable = null, params WordEmbeddingsTransform.Column[] columns)
+            : this(env, new WordEmbeddingsTransform(env, modelKind, customLookupTable, columns))
+        {
+        }
+
+        public WordEmbeddingsEstimator(IHostEnvironment env, WordEmbeddingsTransform transformer)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(WordEmbeddingsEstimator)), transformer)
+        {
+        }
+
+        public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            Host.CheckValue(inputSchema, nameof(inputSchema));
+            var result = inputSchema.Columns.ToDictionary(x => x.Name);
+            foreach (var colInfo in Transformer.Columns)
+            {
+                if (!inputSchema.TryFindColumn(colInfo.input, out var col))
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.input);
+                if (!(col.ItemType is TextType) || col.Kind != SchemaShape.Column.VectorKind.VariableVector)
+                    throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.input, Transformer.InputType.ToString(), col.GetTypeString());
+
+                result[colInfo.output] = new SchemaShape.Column(colInfo.output, SchemaShape.Column.VectorKind.Vector, Transformer.OutputType.ItemType, false);
+            }
+
+            return new SchemaShape(result.Values);
+        }
+    }
+
+    public static class WordEmbeddingsStaticExtensions
+    {
+        public static Vector<float> WordEmbeddings(this Vector<string> input, WordEmbeddingsTransform.PretrainedModelKind modelKind = WordEmbeddingsTransform.PretrainedModelKind.Sswe,
+                string customLookupTable = null)
```

Comment:
As the proposed most conspicuous part of the public API, we have been tending to write clear documentation for our Pigsty extension methods.